src/tools/genpolicy/rules.rego

```diff
@@ -56,13 +54,198 @@ S_NAME_KEY = "io.kubernetes.cri.sandbox-name"
 S_NAMESPACE_KEY = "io.kubernetes.cri.sandbox-namespace"
 BUNDLE_ID = "[a-z0-9]{64}"
 
-CreateContainerRequest:= {"ops": ops, "allowed": true} {
+CreateContainerRequest:= resp {
+    not is_null(input.env_map)
+    
+    i_env_map := input.env_map
+    
+    some p_container in policy_data.containers
+    p_env_map := p_container.env_map  
+    allow_env_map(p_env_map, i_env_map)
+    
+    i_process := input.base.OCI.Process
+    
+    p_process := p_container.OCI.Process
+    
+    allow_args(i_process, p_process, i_env_map)
+
+    resp = CreateContainerRequestCommon(input.base)
+
+    print("PolicyCreateContainerRequest: true")
+}
+
+allow_args(i_process, p_process, i_env_map) {
+    i_args := i_process.Args
+    p_args := p_process.Args
+    print("allow_args: i_args =", i_args, "p_args =", p_args)
+    count(i_args) == count(p_args)
+    every i, i_arg in i_args {
+        print("allow_args: i_arg =", i_arg)
+        p_arg_replaced := replace_env_variables(p_args[i], i_env_map)
+        print("allow_args: p_arg_replaced =", p_arg_replaced)
+        # Double $$ are reduced to a single $, from command field description in https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#container-v1-core
+        p_arg_replaced2 := replace(p_arg_replaced, "$$", "$")
+        print("allow_args: p_arg_replaced2 =", p_arg_replaced2)
+        i_arg == p_arg_replaced2
+        print("allow_args: true")
+    }
+}
+
+allow_args(i_process, p_process, i_env_map) {
+    not i_process.Args
+    not p_process.Args
+    print("allow_args2: no args")
+}
+
+# this function replaces all the environment variables in a string, given a map of environment keys to environment values
+# eg str = "echo $(CLUSTER_ID); echo $(NODE_NAME);"
+# env_map = {"CLUSTER_ID": "abc", "NODE_NAME" : "xyz"}
+# result = "echo abc; echo xyz;"
+replace_env_variables(str, env_map) = result {
+    # make an array of the keys eg ["CLUSTER_ID", "NODE_NAME"]
+    keys := [x | some x in object.keys(env_map)]
+    result := replace_str_rec(str, env_map, keys, count(keys) - 1)
```

Comment:
why do we do result := in this function vs in the functions below result =

Reply:
Good catch. I think we should be using `:=` https://www.openpolicyagent.org/docs/latest/policy-language/#equality-assignment-comparison-and-unification